app/controllers/companies_controller.rb

```diff
@@ -8,4 +8,48 @@ def index
   def show
     @company = Company.find(params[:id])
   end
+
+  def update_db
+    response = read_stock
+    @companies = parse_data(response)
```

Comment:
You could move it to a separate class:

```suggestion
    @companies = GPW::Client.new.fetch_companies
```

and define the class in `lib/gpw/client.rb`.